plugin/web/extensions/geolocation/core/hooks.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from "react";
+
+import { CurrentLocationInfo } from "./types";
+import { postMsg } from "./utils";
+
+const initialLocation: CurrentLocationInfo = {
+  latitude: 5.70249,
+  longitude: 39.7622,
+  altitude: 5000,
+};
+
+const goSuccess = (position: GeolocationPosition) => {
+  return {
+    latitude: position.coords.latitude,
+    longitude: position.coords.longitude,
+    altitude: position.coords.altitude ?? 5000,
+  };
+};
+
+const goError = (err: GeolocationPositionError) => {
+  console.error("Error Code = " + err.code + " - " + err.message);
+  return initialLocation;
+};
+
+const getPosition = () =>
+  new Promise((resolve, reject) =>
+    navigator.geolocation.getCurrentPosition(
+      pos => resolve(goSuccess(pos)),
+      err => reject(goError(err)),
+    ),
+  );
```

Comment:
This code seems a little more than you need.
Check the example [here](https://www.w3schools.com/html/html5_geolocation.asp) for a nicer way of assigning the position to a value

Reply:
done

Reply:
noting that originally I didn't use promise but I wanted to check if the cause any pros or cons here , but obviously same same

Reply:
fc8de4437577d111f90b0a1c7db09d37c48d4de6